spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.springframework.cloud.gcp.data.firestore;
+
+import java.util.Map;
+
+import com.google.cloud.firestore.PublicClassMapper;
+import com.google.firestore.v1.CreateDocumentRequest;
+import com.google.firestore.v1.DeleteDocumentRequest;
+import com.google.firestore.v1.Document;
+import com.google.firestore.v1.FirestoreGrpc;
+import com.google.firestore.v1.RunQueryRequest;
+import com.google.firestore.v1.RunQueryResponse;
+import com.google.firestore.v1.StructuredQuery;
+import com.google.firestore.v1.Value;
+import com.google.protobuf.Empty;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.gcp.data.firestore.mapping.FirestoreMappingContext;
+import org.springframework.cloud.gcp.data.firestore.mapping.FirestorePersistentEntity;
+import org.springframework.cloud.gcp.data.firestore.mapping.FirestorePersistentProperty;
+import org.springframework.cloud.gcp.data.firestore.util.ObservableReactiveUtil;
+import org.springframework.core.annotation.AnnotationUtils;
+
+/**
+ * An implementation of FirestoreReactiveOperations.
+ *
+ * @author Dmitry Solomakha
+ * @since 1.2
+ */
+public class FirestoreTemplate implements FirestoreReactiveOperations {
+	private final FirestoreGrpc.FirestoreStub firestore;
+
+	private final String parent;
+
+	private final FirestoreMappingContext mappingContext = new FirestoreMappingContext();
+
+	public FirestoreTemplate(FirestoreGrpc.FirestoreStub firestore, String parent) {
```

Comment:
May want to have a javadoc here to explain the parameters; I am not quite sure what `parent` refers to.

Reply:
done